core/lib/include/xfa.h

```diff
@@ -138,6 +141,9 @@ _Pragma("GCC diagnostic ignored \"-Warray-bounds\"")
  *
  *     XFA(driver_params, 0) driver_params_t _onboard = { .pin=42 };
  *
+ *
+ * @warning The `driver_params_t` struct must be aligned with the word size
```

Comment:
```suggestion
 * @warning The `driver_params_t` struct must be aligned with the word size.
 *
 * @note    The `WORD_ALIGNED` macro from `#include "architecture.h" provides
 *          the appropriate attribute to ensure word-alignment. Just add this to
 *          the `typedef` statement.
```

Maybe also point the readers to the easiest solution?